services/filemanager/s3manager.go

```diff
@@ -153,53 +162,42 @@ func (manager *S3Manager) getSession() (*session.Session, error) {
 	if manager.Config.Bucket == "" {
 		return nil, errors.New("no storage bucket configured to downloader")
 	}
-
-	getRegionSession := session.Must(session.NewSession())
-	region, err := awsS3Manager.GetBucketRegion(aws.BackgroundContext(), getRegionSession, manager.Config.Bucket, manager.Config.RegionHint)
-	if err != nil {
-		pkgLogger.Errorf("Failed to fetch AWS region for bucket %s. Error %v", manager.Config.Bucket, err)
-		/// Failed to Get Region probably due to VPC restrictions, Will proceed to try with AccessKeyID and AccessKey
+	var region string
+	var err error
+	if manager.Config.Region == nil {
+		getRegionSession := session.Must(session.NewSession())
+		region, err = awsS3Manager.GetBucketRegion(aws.BackgroundContext(), getRegionSession, manager.Config.Bucket, manager.Config.RegionHint)
+		if err != nil {
+			pkgLogger.Errorf("Failed to fetch AWS region for bucket %s. Error %v", manager.Config.Bucket, err)
+			/// Failed to Get Region probably due to VPC restrictions, Will proceed to try with AccessKeyID and AccessKey
+		}
+	} else {
+		region = *manager.Config.Region
 	}
 	var sess *session.Session
 	if manager.Config.AccessKeyID == "" || manager.Config.AccessKey == "" {
 		pkgLogger.Debug("Credentials not found in the destination's config. Using the host credentials instead")
 		sess = session.Must(session.NewSession(&aws.Config{
 			Region:                        aws.String(region),
 			CredentialsChainVerboseErrors: aws.Bool(true),
+			Endpoint:                      manager.Config.Endpoint,
+			S3ForcePathStyle:              manager.Config.S3ForcePathStyle,
+			DisableSSL:                    manager.Config.DisableSSL,
 		}))
 	} else {
 		pkgLogger.Debug("Credentials found in the destination's config.")
 		sess = session.Must(session.NewSession(&aws.Config{
 			Region:                        aws.String(region),
 			Credentials:                   credentials.NewStaticCredentials(manager.Config.AccessKeyID, manager.Config.AccessKey, ""),
 			CredentialsChainVerboseErrors: aws.Bool(true),
+			Endpoint:                      manager.Config.Endpoint,
+			S3ForcePathStyle:              manager.Config.S3ForcePathStyle,
+			DisableSSL:                    manager.Config.DisableSSL,
 		}))
 	}
 	return sess, nil
 }
 
-func (manager *S3Manager) GetDownloadKeyFromFileLocation(location string) string {
-	locationSlice := strings.Split(location, "amazonaws.com/")
-	return locationSlice[len(locationSlice)-1]
-}
-
-/*
-GetObjectNameFromLocation gets the object name/key name from the object location url
-	https://bucket-name.s3.amazonaws.com/key - >> key
-*/
-func (manager *S3Manager) GetObjectNameFromLocation(location string) (string, error) {
-	uri, err := url.Parse(location)
-	if err != nil {
-		return "", err
-	}
-	host := uri.Host
-	path := uri.Path[1:]
-	if strings.Contains(host, manager.Config.Bucket) {
-		return path, nil
-	}
-	return strings.TrimPrefix(path, fmt.Sprintf(`%s/`, manager.Config.Bucket)), nil
```

Comment:
this was probably code for the forcePath option. Kinda dangerous I would say, using `strings.Contains(host, manager.Config.Bucket`  what if the name/key contained the key ?